### PR TITLE
Fix semmle issues

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**
This PR fixes some of the issues that semmle analysis tool found as part of the security review.

We also have some warning in file_utils that I am not sure if/how we should fix:

image

The errors are false-positives after this PR.

### CODE REVIEW

File: src/monitor/formation_metadata.c [hunk 1]
```diff
@@ -745,6 +749,10 @@ AutoFailoverFormationGetDatum(FunctionCallInfo fcinfo, AutoFailoverFormation *fo
 	Datum values[5];
 	bool isNulls[5];
 
+	if (formation == NULL) {
+		return PG_RETURN_DATUM(resultDatum);
```
Comment (JelteF) [inline src/monitor/formation_metadata.c:753]: I think it makes more sense to error out in this case.

File: src/monitor/conninfo.c [hunk 2]
```diff
@@ -52,6 +52,7 @@ ReadPrimaryHostAddress(char **primaryName, char **primaryPort)
 	if (options == NULL)
 	{
 		free(connInfo);
```
Comment (JelteF) [inline src/monitor/conninfo.c:54]: Should this instead be `pfree`?

Comment (SaitTalhaNisanci) [reply src/monitor/conninfo.c:54]: I think so

**Comment by SaitTalhaNisanci** [comment]
Also a note on the function that gives warnings `init_ps_buffer`:

```c
 * Inspired / stolen from Postgres code src/backend/utils/misc/ps_status.c with
 * most of the portability bits removed. At the moment we prefer simple code
 * that works on few targets to highly portable code.
```

so it seems that we took this function from postgres and looking at the function it seems not a critical code as it just sets the process title, so I guess these 2 warnings are okay, what do you think @JelteF?

**Comment by SaitTalhaNisanci** [comment]
image

it seems like I dont' have the restart button for the failed job

File: src/monitor/formation_metadata.c [hunk 3]
```diff
@@ -656,6 +656,11 @@ FormationNumSyncStandbyIsValid(AutoFailoverFormation *formation,
 	List *standbyNodesGroupList = NIL;
 	int count = 0;
 
+	if (formation == NULL)
+	{
+		return false;
```
Comment (JelteF) [inline src/monitor/formation_metadata.c:661]: I think we should error out here as well

Comment (SaitTalhaNisanci) [reply src/monitor/formation_metadata.c:661]: I think we can also add error code for these two errors, probably `errcode(ERRCODE_INVALID_PARAMETER_VALUE)`

**Review by DimCitus — APPROVED** [review]

**Pull request merged**